### PR TITLE
Feature/Create file request

### PR DESCRIPTION
**Pull request by AnnaGakh**
Added Article-File relationship and fileinfo is being sent to File service

### CODE REVIEW

File: src/WikiService.Business/Commands/File/CreateFileCommand.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+      if (response.Errors.Any())
+      {
+        return _responseCreator.CreateFailureResponse<List<Guid>>(HttpStatusCode.BadRequest, response.Errors);
+      }
+
+      response.Body = await _repository.CreateAsync(files.Select(x =>
+        _mapper.Map(x.Id, request.ArticleId)).ToList());
+
+      response.Status = OperationResultStatusType.FullSuccess;
```
Comment (MariyaSin) [inline src/WikiService.Business/Commands/File/CreateFileCommand.cs:96]: sets by default

File: src/WikiService.Business/Commands/File/CreateFileCommand.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      _validator = validator;
+      _bus = bus;
+      _logger = logger;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _responseCreator = responseCreator;
+      _fileDataMapper = fileDataMapper;
+    }
+
+    private async Task<bool> CreateFilesAsync(List<FileData> files, List<string> errors)
```
Comment (MariyaSin) [inline src/WikiService.Business/Commands/File/CreateFileCommand.cs:60]: move private method before ctor

File: src/WikiService.Business/Commands/File/CreateFileCommand.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    {
+      if (files == null || !files.Any())
```
Comment (MariyaSin) [inline src/WikiService.Business/Commands/File/CreateFileCommand.cs:62]: check it in validator

File: src/WikiService.Data/ArticleFileRepository.cs [hunk 4]
```diff
@@ -1,16 +1,34 @@
...
+
+        public ArticleFileRepository(
+          IDataProvider provider)
+        {
+            _provider = provider;
+        }
+
+        public async Task<List<Guid>> CreateAsync(List<DbArticleFile> files)
+        {
+            if (files == null)
```
Comment (MariyaSin) [inline src/WikiService.Data/ArticleFileRepository.cs:23]: or !files.Any()

File: src/WikiService.Mappers/Models/FileDataMapper.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  public class FileDataMapper : IFileDataMapper
+  {
+    public FileData Map(FileInfo file)
+    {
+      if (file is null)
+      {
+        return null;
+      }
+
+      Guid fileId = Guid.NewGuid();
```
Comment (MariyaSin) [inline src/WikiService.Mappers/Models/FileDataMapper.cs:17]: no need create the variable, set Guid.NewGuid() in 20 line

File: src/WikiService.Models.Dto/Requests/File/CreateFileRequest.cs [hunk 6]
```diff
@@ -1,19 +1,11 @@
...
-  public record CreateFileRequest
-  {
-    [Required]
-    public string Name { get; set; }
-    [Required]
-    public string Content { get; set; }
-    [Required]
-    public string Extension { get; set; }
-  }
+    public record CreateFileRequest
```
Comment (MariyaSin) [inline src/WikiService.Models.Dto/Requests/File/CreateFileRequest.cs:6]: CreateArticleFilesRequest

File: src/WikiService.Validation/File/CreateFileRequestValidator.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace LT.DigitalOffice.WikiService.Validation.File
+{
+  public class CreateFileRequestValidator : AbstractValidator<CreateFileRequest>, ICreateFileRequestValidator
+  {
+    public CreateFileRequestValidator(
+      IArticleRepository articleRepository)
+    {
+      RuleFor(file => file.ArticleId)
+      .MustAsync(async (articleId, _) => await articleRepository.DoesExistAsync(articleId))
```
Comment (MariyaSin) [inline src/WikiService.Validation/File/CreateFileRequestValidator.cs:14]: fix tab

File: src/WikiService.Validation/File/CreateFileRequestValidator.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+      .WithMessage("This article id does not exist");
```
Comment (MariyaSin) [inline src/WikiService.Validation/File/CreateFileRequestValidator.cs:15]: add files check

File: src/WikiService.Models.Dto/Requests/File/CreateFileRequest.cs [hunk 9]
```diff
@@ -1,19 +1,11 @@
...
+    {
+        public Guid ArticleId { get; set; }
+        public List<FileInfo> Files { get; set; }
```
Comment (MariyaSin) [inline src/WikiService.Models.Dto/Requests/File/CreateFileRequest.cs:9]: add [Requred]

File: src/WikiService.Validation/File/CreateFileRequestValidator.cs (hunk 8 above)
Comment (MariyaSin) [reply src/WikiService.Validation/File/CreateFileRequestValidator.cs:15]: check if files list is empty

File: src/WikiService/Controllers/FileController.cs [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.WikiService.Controllers
+{
+  [Route("[controller]")]
+  [ApiController]
+  public class FileController : ControllerBase
+  {
+    [HttpPost("create")]
+    public async Task<OperationResultResponse<List<Guid>>> CreateAsync(
```
Comment (MariyaSin) [inline src/WikiService/Controllers/FileController.cs:16]: return bool

File: src/WikiService/Startup.cs [hunk 11]
```diff
@@ -137,7 +137,7 @@ private void UpdateDatabase(IApplicationBuilder app)
 
       using var context = serviceScope.ServiceProvider.GetService<WikiServiceDbContext>();
 
-      context.Database.Migrate();
+      //context.Database.Migrate();
```
Comment (MariyaSin) [inline src/WikiService/Startup.cs]: ?

File: src/WikiService.Business/Commands/File/CreateFileCommand.cs [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IResponseCreator _responseCreator;
+    private readonly IFileDataMapper _fileDataMapper;
+
+    public CreateFileCommand(
+      IDbArticleFileMapper mapper,
+      IArticleFileRepository repository,
+      ICreateFileRequestValidator validator,
+      IBus bus,
+      ILogger<CreateFileCommand> logger,
```
Comment (MariyaSin) [inline src/WikiService.Business/Commands/File/CreateFileCommand.cs:43]: do you use ligger?

File: src/WikiService.Business/Commands/File/CreateFileCommand.cs [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+      {
+        return false;
+      }
+
+      await _bus.Publish<ICreateFilesPublish>(ICreateFilesPublish.CreateObj(
+        files, _httpContextAccessor.HttpContext.GetUserId()));
+
+      return true;
+    }
+    public async Task<OperationResultResponse<List<Guid>>> ExecuteAsync(CreateFileRequest request)
```
Comment (ndoriomedov) [inline src/WikiService.Business/Commands/File/CreateFileCommand.cs:72]: add empty line

File: src/WikiService.Business/Commands/File/CreateFileCommand.cs [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    {
+      ValidationResult validationResult = await _validator.ValidateAsync(request);
+
+      if (!validationResult.IsValid)
+      {
+        return _responseCreator.CreateFailureResponse<List<Guid>>(
+          HttpStatusCode.BadRequest,
+          validationResult.Errors.Select(vf => vf.ErrorMessage).ToList());
+      }
```
Comment (ndoriomedov) [inline src/WikiService.Business/Commands/File/CreateFileCommand.cs:81]: add empty line after this

File: src/WikiService.Business/Commands/File/CreateFileCommand.cs [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+      OperationResultResponse<List<Guid>> response = new();
+
+      List<FileData> files = request.Files?.Select(x => _fileDataMapper.Map(x)).ToList();
+
+      await CreateFilesAsync(files, response.Errors);
+
+      if (response.Errors.Any())
+      {
+        return _responseCreator.CreateFailureResponse<List<Guid>>(HttpStatusCode.BadRequest, response.Errors);
+      }
```
Comment (ndoriomedov) [inline src/WikiService.Business/Commands/File/CreateFileCommand.cs:91]: will response.Errors ever be not empty?

File: src/WikiService.Data/ArticleFileRepository.cs (hunk 4 above)
Comment (ndoriomedov) [reply src/WikiService.Data/ArticleFileRepository.cs:23]: and use 'is' instead of '==' in null checks

**Pull request closed without merging**